pkg/volume/git_repo/git_repo.go

```diff
@@ -138,8 +140,15 @@ func (b *gitRepoVolumeBuilder) SetUpAt(dir string) error {
 		return err
 	}
 
-	if output, err := b.execCommand("git", []string{"clone", b.source}, dir); err != nil {
-		return fmt.Errorf("failed to exec 'git clone %s': %s: %v", b.source, output, err)
+	// If provided target directory
+	if "" != b.target {
+		if output, err := b.execCommand("git", []string{"clone", b.source, b.target}, dir); err != nil {
+			return fmt.Errorf("failed to exec 'git clone %s %s': %s: %v", b.source, b.target, output, err)
+		}
+	} else {
+		if output, err := b.execCommand("git", []string{"clone", b.source}, dir); err != nil {
+			return fmt.Errorf("failed to exec 'git clone %s': %s: %v", b.source, output, err)
+		}
 	}
```

Comment:
Will this part fail if you specify a target?  The current implementation assumes there will be exactly 1 directory (the repo name) in the volume.

```
if len(files) != 1 {
 		return fmt.Errorf("unexpected directory contents: %v", files)
 	}
```

I couldn't comment on that line in the diff because you didn't edit it.